src/CattyTests/PenFeatureTests.swift

```diff
@@ -0,0 +1,254 @@
+/**
+ *  Copyright (C) 2010-2020 The Catrobat Team
+ *  (http://developer.catrobat.org/credits)
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  An additional term exception under section 7 of the GNU Affero
+ *  General Public License, version 3, is available at
+ *  (http://developer.catrobat.org/license_additional_term)
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+
+import XCTest
+
+@testable import Pocket_Code
+
+final class PenFeatureTests: XCTestCase {
+
+    var project: ProjectMock!
+
+    var spriteObject: SpriteObject!
+    var initialPosition: CGPoint!
+
+    override func setUp() {
+        self.project = ProjectMock()
+
+        self.spriteObject = SpriteObject()
+        spriteObject.name = "testName"
+        self.initialPosition = CGPoint.zero
+        self.project.objectList.add(spriteObject as Any)
+    }
+
+    func testWhenSpritePositionChangedPenUp() {
+
+        let scene = SceneBuilder(project: self.project).build()
+        let sceneStarted = scene.startProject()
+        XCTAssertEqual(sceneStarted, true)
+
+        guard let spriteNode = scene.childNode(withName: "testName") as? CBSpriteNode else {
+            XCTAssertEqual(1, 0) // Test fails when spriteNode is nil
```

Comment:
Please use `XCTFail` with a meaningful message so that it is clear why the test fails. You can then remove your comment.